### PR TITLE
Admin Merchants Index -- Top Five Merchants

### LINKED ISSUE #11: Admin Merchants: Top Merchant's Best Day
As an admin,
When I visit the admin merchants index
Then next to each of the 5 merchants by revenue I see the date with the most revenue for each merchant.
And I see a label “Top selling date for <merchant name> was <date with most sales>"

Note: use the invoice date. If there are multiple days with equal number of sales, return the most recent day.

### LINKED ISSUE #12: Top 5 Merchants By Revenue
As an admin,
When I visit the admin merchants index
Then I see the names of the top 5 merchants by total revenue generated
And I see that each merchant name links to the admin merchant show page for that merchant
And I see the total revenue generated next to each merchant name

Notes on Revenue Calculation:
- Only invoices with at least one successful transaction should count towards revenue
- Revenue for an invoice should be calculated as the sum of the revenue of all invoice items
- Revenue for an invoice item should be calculated as the invoice item unit price multiplied by the quantity (do not use the item unit price)

### PR DESCRIPTION
**Pull request by aschwartz1**
**What Changed**
- Added many things to `Merchant`
  - `::top_five_merchants`
  - `#total_revenue`
  - `#best_day`
- Added 'Top Merchants' section to the Admin Merchants Index
  - Includes list of top 5 merchants by revenue, closes #12
  - includes best day for each of top 5 merchants, closes #11

**Next Steps**
- This finishes all of the Admin Merchants user stories. Up next is the API work and any refactoring?

**Known Issues**
- Test setup for this made already-long setups even longer. There _has_ to be a better way to organize this stuff... but red/green/refactor, right?  Right...

### CODE REVIEW

File: spec/features/admin/merchants/index_spec.rb [hunk 1]
```diff
@@ -85,5 +83,218 @@
       click_link('New Merchant')
       expect(current_path).to eq(new_admin_merchant_path)
     end
+
+    describe 'top merchants section' do
+      it 'shows top five merchants by revenue' do
+        setup_top_five_revenue_merchants
+        visit admin_merchants_path
```
Comment (amsmyth1) [inline spec/features/admin/merchants/index_spec.rb:90]: This was a great idea to have different setups you call.

File: spec/features/admin/merchants/index_spec.rb [hunk 2]
```diff
@@ -85,5 +83,218 @@
...
+
+  def setup_basic_merchants
+    @merchant_11 = create(:merchant, status: :enabled)
+    @merchant_12= create(:merchant, status: :enabled)
+    @merchant_13 = create(:merchant, status: :enabled)
+    @merchant_14, @merchant_15, @merchant_16 = create_list(:merchant, 3)
+  end
+
+  def setup_top_five_revenue_merchants
+    merchant_1_with_history
```
Comment (amsmyth1) [inline spec/features/admin/merchants/index_spec.rb:171]: This test setup is so organized, great idea to make this lengthy file work so well.

**Review by amsmyth1 — APPROVED** [review]
Everything looks great!

**Comment by amsmyth1** [comment]
Great use of helper setup methods to test all edge cases of the top_merchants.
